src/sidebar/components/thread.js

```diff
@@ -0,0 +1,116 @@
+import classnames from 'classnames';
+import { createElement, Fragment } from 'preact';
+
+import propTypes from 'prop-types';
+import useStore from '../store/use-store';
+import { withServices } from '../util/service-context';
+import { countHidden, countVisible } from '../util/thread';
+
+import Annotation from './annotation';
+import Button from './button';
+import ModerationBanner from './moderation-banner';
+
+/**
+ * A thread, which contains a single annotation at its top level, and its
+ * recursively-rendered children (i.e. replies). A thread may have a parent,
+ * and at any given time it may be `collapsed`.
+ */
+function Thread({ showDocumentInfo = false, thread, threadsService }) {
+  const setCollapsed = useStore(store => store.setCollapsed);
+
+  // Only render this thread's annotation if it exists and the thread is `visible`
+  const showAnnotation = thread.annotation && thread.visible;
```

Comment:
I confess to still not being entirely clear what `visible` really means and where it gets mutated. My hunch is that `visible` threads are on-screen and, well, visible.

Reply:
The `Thread` data structure and the `visible` property is defined here: https://github.com/hypothesis/client/blob/d4bd1896fcf47450c36134ef60c0a0ca2234596a/src/sidebar/build-thread.js#L21

FWIW the preferred approach would be to define a JSDoc `@typedef` for a `Thread` and then reference it elsewhere to make this easier to look up.

Reply:
Hmmm, by the time `thread`s get to this component, they've got additional properties than those listed here, e.g. `depth` and `replyCount`. I'll do more digging.